src/session/content/item_row.rs

```diff
@@ -94,16 +96,311 @@ impl ItemRow {
+                                                gettext!("{} changed group photo", sender_name);
+                                            child.set_label(&format!("<b>{}</b>", msg))
+                                        }
+                                    }
+                                }
+                            }
+                            MessageContent::MessageChatDeletePhoto => {
+                                let child = self.get_or_create_event_row();
+                                let type_ = message.chat().type_().clone();
+
+                                match type_ {
+                                    ChatType::Supergroup(data) => {
+                                        let is_channel = data.is_channel();
+                                        if is_channel {
+                                            let msg = gettext("Channel photo removed");
+                                            child.set_label(&format!("<b>{}</b>", msg))
+                                        } else if message.is_outgoing() {
+                                            let msg = gettext("You removed group photo");
+                                            child.set_label(&format!("<b>{}</b>", msg))
+                                        } else {
+                                            let msg =
+                                                gettext!("{} changed group photo", sender_name);
+                                            child.set_label(&format!("<b>{}</b>", msg))
+                                        }
+                                    }
+                                    _ => {
+                                        if message.is_outgoing() {
+                                            let msg = gettext("You removed group photo");
+                                            child.set_label(&format!("<b>{}</b>", msg))
+                                        } else {
+                                            let msg =
+                                                gettext!("{} removed group photo", sender_name);
+                                            child.set_label(&format!("<b>{}</b>", msg))
+                                        }
+                                    }
+                                }
+                            }
+                            MessageContent::MessageChatJoinByRequest => {
+                                let child = self.get_or_create_event_row();
+                                if message.is_outgoing() {
+                                    let msg = gettext!("You joined the group", sender_name);
+                                    child.set_label(&format!("<b>{}</b>", msg));
+                                } else {
+                                    let msg = gettext!("{} joined the group", sender_name);
+                                    child.set_label(&format!("<b>{}</b>", msg));
+                                }
```

Comment:
Could you change it to this? This not only makes it shorter, but it also makes the intention much more clear.
This also applies to the next match arm.
```suggestion
                                child.set_label(&format!(
                                    "<b>{}</b>",
                                    if message.is_outgoing() {
                                        gettext!("You joined the group", sender_name)
                                    } else {
                                        gettext!("{} joined the group", sender_name)
                                    }
                                ));
```